server/src/main/java/io/crate/expression/reference/doc/lucene/SourceLookup.java

```diff
@@ -115,6 +115,9 @@ static Object extractValue(final Map<?, ?> map, List<String> path, int pathStart
                 }
                 return newList;
             } else {
+                if (i + 1 != path.size()) {
+                    return null;
+                }
                 break;
```

Comment:
I wonder if we can remove `SourceLookup.extractValue()` completely and use `ValueExtractors.fromMap()` instead.

`ValueExtractors.fromMap` internally uses `Maps.getyPath `which seems to cover already this `if (i + 1 == path.size()) {`

PKLookup already uses `ValueExtractors` to deal with parsed source (we provide parsed source, processed by SourceParser in https://github.com/crate/crate/pull/14697)

Reply:
Forgot to mention, if we use ValueExtractors and it doesn't pass your test, we can fix ValueExtractors itself and such fix not only SourceLookup but PKLookup as well.

Reply:
Sounds good to deduplicate some of that, but I'd do that as a follow up. For a fix that gets backported, a small fix like this is ideal

Reply:
I'm merging it now then - about to rebase drop-column, SourceParser is actively updated there, let's get this change asap.